src/JustSaying/JustSayingBus.cs

```diff
@@ -115,27 +139,43 @@ private void AddSubscribersToInterrogationResponse(INotificationSubscriberInterr
             publishersByType[topicType] = messagePublisher;
         }
 
-        public void Start(CancellationToken cancellationToken = default)
+        public Task Start(CancellationToken cancellationToken)
         {
             if (cancellationToken.IsCancellationRequested)
             {
-                return;
+                return Task.CompletedTask;
+            }
+
+            if (ConsumerBus != null)
+            {
+                _log.LogWarning("Attempting to start an already running Bus");
+                return ConsumerBus.Run(CancellationToken.None);
```

Comment:
Why isn't `cancellationToken` passed through?

Reply:
This is relying on the fact that calling `ConsumerBus.Run` multiple times will always return the same task, which perhaps isn't very clear. Only the first time will it set the cancellation token, which happens below.

I'm changing this to be explicit about the single-start safety by keeping a `Task _consumerCompletionTask` that we return on multiple-entries to this bus.